src/components/app.tsx

```diff
@@ -101,23 +103,18 @@ export default function App(props: AppProps): JSX.Element {
   const trajManager = React.useRef<RobotTrajectoryManager | undefined>(undefined);
 
   const doorStateManager = React.useMemo(() => new DoorStateManager(), []);
-  const [doorStates, setDoorStates] = React.useState<Readonly<Record<string, RomiCore.DoorState>>>(
-    {},
-  );
+  const [doorStates, setDoorStates] = React.useState(doorStateManager.doorStates());
   const [doors, setDoors] = React.useState<readonly RomiCore.Door[]>([]);
-  // FIXME: not used for now as there is not enough information to render doors.
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+
   const [doorSpotlight, setDoorSpotlight] = React.useState<SpotlightValue<string> | undefined>(
     undefined,
   );
 
   const liftStateManager = React.useMemo(() => new LiftStateManager(), []);
-  const [liftStates, setLiftStates] = React.useState<Readonly<Record<string, RomiCore.LiftState>>>(
-    {},
-  );
-  const [lifts, setLifts] = React.useState<readonly RomiCore.Lift[]>([]);
-  // FIXME: not used for now as there is not enough information to render lifts.
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [liftStates, setLiftStates] = React.useState(liftStateManager.liftStates());
```

Comment:
```suggestion
  const [liftStates, setLiftStates] = React.useState(() => liftStateManager.liftStates());
```
`liftStateManager.liftStates()` creates a shallow copy of all the current states, which may be somewhat expensive so its better to use lazy initial state.

Reply:
Yeah! you're totally right! Thank you!